src/DelayDiffEq.jl

```diff
@@ -5,34 +5,31 @@ module DelayDiffEq
 using Reexport
 @reexport using OrdinaryDiffEq
 
-using DataStructures, RecursiveArrayTools, Combinatorics, MuladdMacro, ForwardDiff, NLsolve
+using DataStructures, RecursiveArrayTools, MuladdMacro, ForwardDiff, NLsolve, Roots
 
 import OrdinaryDiffEq: initialize!, perform_step!, loopfooter!, loopheader!, alg_order,
-                       handle_tstop!, ODEIntegrator, savevalues!,
-                       handle_callback_modifiers!
+                       handle_tstop!, ODEIntegrator, savevalues!, postamble!,
+                       handle_callback_modifiers!, reeval_internals_due_to_modification!,
+                       handle_discontinuities!, find_callback_time, apply_callback!
 
 import DiffEqBase: solve, solve!, init, resize!, u_cache, user_cache, du_cache, full_cache,
                    deleteat!, terminate!, u_modified!, get_proposed_dt, set_proposed_dt!
 
 import OrdinaryDiffEq: Rosenbrock23Cache, Rosenbrock32Cache, ImplicitEulerCache,
                        TrapezoidCache
 
+include("discontinuity_type.jl")
 include("integrator_type.jl")
 include("integrator_utils.jl")
 include("integrator_interface.jl")
+include("interpolants.jl")
 include("history_function.jl")
 include("algorithms.jl")
 include("callbacks.jl")
 include("alg_utils.jl")
 include("solve.jl")
 include("utils.jl")
 
-export init
-
-export initialize!, loopheader!, perform_step!, loopfooter!, handle_tstop!, postamble!
-
-export HistoryFunction
-
-export MethodOfSteps
+export Discontinuity, MethodOfSteps
```

Comment:
Does Discontinuity have to be exported?

Reply:
Disontinuities have to be specified as vector of `Discontinuity` in `solve`, so yes. We could allow simple time vectors as well and cnvert those time points to discontinuities of order 0, but even then I think it is still good to be able to specify concrete discontinuities sometimes.